tests/unit/charms/tls_certificates_interface/v1/test_tls_certificates_v1_requires.py

```diff
@@ -680,4 +681,167 @@ def test_given_csr_in_unit_relation_data_and_certificate_revoked_in_remote_relat
             key_values=remote_app_relation_data,
         )
 
-        patch_on_certificate_revoked.assert_not_called()
+        patch_on_certificate_invalidated.assert_not_called()
+
+    @patch(f"{BASE_CHARM_DIR}._on_all_certificates_invalidated")
+    def test_given_certificate_in_relation_data_when_relation_broken_then_all_certificates_invalidated_event_is_emitted(  # noqa: E501
+        self, pach_on_all_certificates_invalidated
+    ):
+        relation_id = self.create_certificates_relation()
+        ca_certificate = "whatever certificate"
+        chain = ["certificate 1", "certiicate 2", "certificate 3"]
+        csr = "whatever csr"
+        certificate = "whatever certificate"
+        unit_relation_data = {
+            "certificate_signing_requests": json.dumps([{"certificate_signing_request": csr}])
+        }
+        remote_app_relation_data = {
+            "certificates": json.dumps(
+                [
+                    {
+                        "ca": ca_certificate,
+                        "chain": chain,
+                        "certificate_signing_request": csr,
+                        "certificate": certificate,
+                        "revoked": False,
+                    }
+                ]
+            )
+        }
+
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=self.harness.charm.unit.name,
+            key_values=unit_relation_data,
+        )
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=self.remote_app,
+            key_values=remote_app_relation_data,
+        )
+        self.harness.remove_relation(relation_id)
```

Comment:
We should add the 2 following tests:
- A test that validates the reason associated to a relation being removed
- A test that validates the behavior when the relation is removed given there are no certificates in the relation data.

Reply:
Why we care about "the reason associated to a relation being removed", if the relation is removed, we want to invalidate all certs, don't we? The only case in which I see we don't, it would be when there are 2 or more units and one of them is removed, so we don't want to scrap all certs because one is gone. Let me know your thoughts please.
I've added the second one.

Reply:
I was under the impression we had a reason associated to `AllCertificatesInvalidatedEvent` but indeed we don't. You can ignore the first one then

Reply:
Nice, then It's done.